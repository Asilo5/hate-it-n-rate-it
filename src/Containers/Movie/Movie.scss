.movie {
    width: 65%;
    display: flex;
    // align-items: center;
    margin: 10px auto;
    position: relative;
    top: 3em;
    img {
        width: 40%;
    }
}