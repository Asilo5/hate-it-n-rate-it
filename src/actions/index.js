export const addMoviesData = movies => ({
  type: 'ADD_MOVIES_DATA',
  movies
})

export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const hasError = (error) => ({
  type: 'HAS_ERROR',
  error
})

