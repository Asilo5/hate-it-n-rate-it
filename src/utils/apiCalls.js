export const fetchUser = async (email, password) => {
  const userInfo = {email, password};
  const options = {
    method: 'POST',
    body: JSON.stringify(userInfo),
    headers: {'Content-Type': 'application/json'}
  }
  const response = await fetch('https://rancid-tomatillos.herokuapp.com/api/v1/login', options);
  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.message);
  }
  const user = await response.json();
  return user;
}

export const fetchRatings = async () => {
  const response = await fetch('https://rancid-tomatillos.herokuapp.com/api/v1/users/5/ratings');
  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.message);
  }
  const movieRatings = await response.json();
  console.log(movieRatings);
  return movieRatings;
}