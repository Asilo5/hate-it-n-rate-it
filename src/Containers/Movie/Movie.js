import React, { Component } from 'react';
import { addRatings, getMovies, fetchRatings } from '../../utils/apiCalls';
import { addMoviesData, setUserRatings } from '../../actions';
import './Movie.scss';
import { connect } from 'react-redux';


export class Movie extends Component {
    constructor(props) {
        super(props);
        this.state = {
            rating: null
        }
    }

    handleChange = (value) => {
        this.setState({rating: parseInt(value)})
    }

    handleSubmit = async () => {
        addRatings(this.props.id, this.state.rating);
        const movies = await getMovies();
        this.props.addMoviesData(movies);
        fetchRatings()
            .then(data => this.props.setUserRatings(data))
    }
    
    findRatedMovie = () => {
        if (this.props.userRatings.length !== 0) {
            return this.props.userRatings.find(movie => movie.movie_id === parseInt(this.props.id))
        }
    }
    
    render() {
        const { average_rating, id, backdrop_path, overview, poster_path, release_date, title, userRatings} = this.props

        return ( 
            <section className='movie'>
                <h2 className='title' >{title}</h2>
                <h3>Out on: {release_date}</h3>
                <p>Avg. rating: {average_rating.toFixed(2)}</p>
                {this.findRatedMovie() ? 
                    <p>Your rating: {this.findRatedMovie().rating}</p>
                :
                    <div className='rating'>
                        <p> Add your rating </p>
                        <form>
                            <select onChange={(e) => this.handleChange(e.target.value)}>
                                <option value={1} >1</option>
                                <option value={2} >2</option>
                                <option value={3} >3</option>
                                <option value={4} >4</option>
                                <option value={5} >5</option>
                                <option value={6} >6</option>
                                <option value={7} >7</option>
                                <option value={8} >8</option>
                                <option value={9} >9</option>
                                <option value={10} >10</option>
                            </select>
                            <button 
                                type='button'
                                onClick={this.handleSubmit}
                                >SUBMIT
                            </button>
                        </form>
                    </div>
                }
                <img className ='back-ground' src={backdrop_path} alt={title} />
                <section className='movie-info'>
                    <img className ='main-img' src={poster_path} alt={title} />
                    <div className='overview-rating'>
                        <p>{overview}</p>
                    </div>
                </section>
            </section>
        )
    }
} 

const mapDispatchToProps = dispatch => ({
    addMoviesData: movies => dispatch(addMoviesData(movies)),
    setUserRatings: ratedMovies => dispatch(setUserRatings(ratedMovies))
  })

export const mapStateToProps = (state) => ({
    movies: state.movies,
    userRatings: state.userRatings,
    user: state.user
})

export default connect(mapStateToProps, mapDispatchToProps)(Movie);

