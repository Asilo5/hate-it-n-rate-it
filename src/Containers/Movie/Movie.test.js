import React from 'react';
import { Movie, mapDispatchToProps } from './Movie';
import { shallow } from 'enzyme';
import { addMoviesData, setUserRatings } from '../../actions';
import * as mockData from '../../utils/mockData';

describe('Movie', () => {
  it('should match the snapshot', () => {
    const mockAverage = 4;
    const mockRatings = [];
    const wrapper = shallow(<Movie userRatings={mockRatings} average_rating={mockAverage} />);
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a addMoviesData action', () => {
     const mockDispatch = jest.fn();
     const actionToDispatch = addMoviesData(mockData.mockMovies);
     const mappedProps = mapDispatchToProps(mockDispatch);
 
     mappedProps.addMoviesData(mockData.mockMovies);
 
     expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with a setUserRatings action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setUserRatings(mockData.mockRatedMovies);
      const mappedProps = mapDispatchToProps(mockDispatch);
  
      mappedProps.setUserRatings(mockData.mockRatedMovies);
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
     })
 
  })

})