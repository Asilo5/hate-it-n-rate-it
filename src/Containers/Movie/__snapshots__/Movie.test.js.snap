// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match the snapshot 1`] = `
<section
  className="movie"
>
  <h2
    className="title"
  />
  <h3>
    Out on: 
  </h3>
  <p>
    Avg. rating: 
  </p>
  <img
    className="back-ground"
  />
  <section
    className="movie-info"
  >
    <img
      className="main-img"
    />
    <div
      className="overview-rating"
    >
      <p />
      <div
        className="rating"
      >
        <p>
           Add your rating 
        </p>
      </div>
    </div>
  </section>
</section>
`;
